zookeeper-server/src/main/java/org/apache/zookeeper/MultiResponse.java

```diff
@@ -78,6 +82,12 @@ public void serialize(OutputArchive archive, String tag) throws IOException {
                 case ZooDefs.OpCode.setData:
                     new SetDataResponse(((OpResult.SetDataResult) result).getStat()).serialize(archive, tag);
                     break;
+                case ZooDefs.OpCode.getChildren:
+                    new GetChildrenResponse(((OpResult.GetChildrenResult) result).getChildren()).serialize(archive, tag);
+                    break;
+                case ZooDefs.OpCode.getData:
+                    new GetDataResponse(((OpResult.GetDataResult) result).getData(), new Stat()).serialize(archive, tag);
```

Comment:
Let's return Stat from the GetData call:

new GetDataResponse(((OpResult.GetDataResult) result).getData(),((OpResult.GetDataResult) result).getStat()).serialize(archive, tag);